### PR TITLE
Detect and update errata when not all repository packages are linked

### PR DESCRIPTION
**Pull request by wweellddeerr**
## What does this PR change?

This PR updates the logic for determining whether an errata requires an update. It introduces a new check that compares the packages associated with the errata in the repository against those in the database. If any package is found in the repository but is not yet reflected in the database, the errata is flagged for an update.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: bug fix

- [x] **DONE**

## Test coverage
ℹ️ If a major new functionality is added, it is **strongly recommended** that tests for the new functionality are added to the Cucumber test suite
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/24793

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: python/spacewalk/satellite_tools/reposync.py [hunk 1]
```diff
@@ -1192,19 +1192,27 @@ def _populate_erratum(self, notice):
             updated_date = self._to_db_date(notice["updated"])
         else:
             updated_date = self._to_db_date(notice["issued"])
+
+        existing_packages = existing_errata["packages"] if existing_errata else []
+        npkgs = [pkg for c in notice["pkglist"] for pkg in c["packages"]]
```
Comment (agraul) [inline python/spacewalk/satellite_tools/reposync.py:1197]: Please don't use list comprehensions with multiple for clauses ([style guide](https://google.github.io/styleguide/pyguide.html#27-comprehensions--generator-expressions)).

File: python/spacewalk/satellite_tools/reposync.py [hunk 2]
```diff
@@ -1192,19 +1192,27 @@ def _populate_erratum(self, notice):
...
+
+        e = importLib.Erratum()
+        e["packages"] = self._updates_process_packages(
+            npkgs, e["advisory_name"], list(existing_packages)
```
Comment (agraul) [inline python/spacewalk/satellite_tools/reposync.py:1201]: Isn't `existing_packages` always a list?

File: python/spacewalk/satellite_tools/reposync.py [hunk 3]
```diff
@@ -2546,12 +2547,13 @@ def upload_patches(self, notices):
         return processed_updates
 
     def errata_needs_update(
-        self, existing_errata, new_errata_version, new_errata_changedate
+        self, existing_errata, new_errata_version, new_errata_changedate, packages=[]
```
Comment (agraul) [inline python/spacewalk/satellite_tools/reposync.py:2550]: A list is not a good default argument. Please use `None` and check for `None` inside the function body to set it to an empty list instead ([style guide](https://google.github.io/styleguide/pyguide.html#212-default-argument-values)).

**Review by agraul — COMMENTED** [review]
When I read the PR description I thought you added a new method that does the check and wondered why the existing methods weren't used. Then I realized you use (one of) them and just moved it around. 
Can you add a code comment explaining why it has to be done this early?

Oh, and is this also needed for `notices_type == 'patches`?

**Comment by ycedres** [comment]
@wweellddeerr just a question, so the errata can be reissued with the same version, same change date an different set of affected packages?

File: python/spacewalk/satellite_tools/reposync.py (hunk 2 above)
Comment (agraul) [reply python/spacewalk/satellite_tools/reposync.py]: Oh, I didn't consider that you wanted to create a copy. In that case I suggest to use `existing_packages.copy()`, it's more explicit.

**Review by ycedres — APPROVED** [review]

**Pull request merged**